src/gewis/service/gewisdb-service.ts

```diff
@@ -0,0 +1,128 @@
+/**
+ *  SudoSOS back-end API service.
+ *  Copyright (C) 2020  Study association GEWIS
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as published
+ *  by the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+import GewisUser from '../entity/gewis-user';
+import { BasicApi, Configuration, Health, MembersApi } from 'gewisdb-ts-client';
+import log4js, { Logger } from 'log4js';
+import { webResponseToUpdate } from '../helpers/gewis-helper';
+import UserService from '../../service/user-service';
+import { UserResponse } from '../../controller/response/user-response';
+
+const GEWISDB_API_URL = process.env.GEWISDB_API_URL;
+const GEWISDB_API_KEY = process.env.GEWISDB_API_KEY;
+
+// Configuration for the API access
+const configuration = new Configuration({ basePath: GEWISDB_API_URL, accessToken: () => GEWISDB_API_KEY });
+const api = new MembersApi(configuration);
+const pinger = new BasicApi(configuration);
+
+// Logger setup
+const logger: Logger = log4js.getLogger('GewisDBService');
+logger.level = process.env.LOG_LEVEL;
+
+export default class GewisDBService {
+
+  public static api = api;
+
+  public static pinger = pinger;
+
+  /**
+   * Synchronizes ALL users with GEWIS DB user data.
+   * This method only returns users that were actually updated during the synchronization process.
+   * @returns {Promise<UserResponse[]>} A promise that resolves with an array of UserResponses for users that were updated. Returns null if the API is unhealthy.
+   */
+  public static async syncAll(): Promise<UserResponse[]> {
+    const gewisUsers = await GewisUser.find({ where: { user: { deleted: false } }, relations: ['user'] });
+    return this.sync(gewisUsers);
+  }
+
+  /**
+   * Synchronizes users with GEWIS DB user data.
+   * This method only returns users that were actually updated during the synchronization process.
+   * @param {GewisUser[]} gewisUsers - Array of users to sync.
+   * @returns {Promise<UserResponse[]>} A promise that resolves with an array of UserResponses for users that were updated. Returns null if the API is unhealthy.
+   */
+  public static async sync(gewisUsers: GewisUser[]): Promise<UserResponse[]> {
+    const ping: Health = await GewisDBService.pinger.healthGet().then(member => member)
+      .catch((error) => {
+        logger.warn('Failed to ping GEWIS DB', error);
+        return null;
```

Comment:
Because you return null in the `.catch` block, this block returns null but the execution flow of the `sync` function will continue. This will then cause a crash later because `ping` is null. Maybe wrap the await in a try-catch block.